.github/workflows/ci.yml

```diff
@@ -0,0 +1,137 @@
+      ALGOLIA_INDEX_NAME: ${{ vars.ALGOLIA_INDEX_NAME }}
+      SANDBOX_HOST: ${{ vars.SANDBOX_HOST }}
+      TEST_PRODUCT_ID: ${{ vars.TEST_PRODUCT_ID }}
+      SFCC_OAUTH_CLIENT_ID: ${{ secrets.SFCC_OAUTH_CLIENT_ID }}
+      SFCC_OAUTH_CLIENT_SECRET: ${{ secrets.SFCC_OAUTH_CLIENT_SECRET }}
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+
+      - name: Install Required Packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y zip
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Lint JavaScript
+        run: npm run lint:js
+
+      - name: Run Unit Tests
+        run: npm run test
+
+      - name: Compile JS and CSS
+        run: |
+          npm run compile:js
+          npm run compile:scss
+
+      - name: Create Code Version Directory
+        run: mkdir $CODE_VERSION
+
+      - name: Copy Cartridges
+        run: cp -R cartridges/* $CODE_VERSION/
+
+      - name: Package Code
+        run: zip -r $CODE_VERSION.zip $CODE_VERSION
+
+      - name: Deploy and Activate Code
+        env:
+          SANDBOX_HOST: ${{ vars.SANDBOX_HOST }}
+          CODE_VERSION: ${{ vars.CODE_VERSION }}
+        run: npm run deploy:code
+
+      - name: Import Site Preferences
+        env:
+          SANDBOX_HOST: ${{ vars.SANDBOX_HOST }}
+        run: npm run import:preferences
+
+      - name: Run SFCC Job
+        env:
+          SANDBOX_HOST: ${{ vars.SANDBOX_HOST }}
+        run: npm run run:sfcc-job
+
+  unit-tests:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Run Unit Tests
+        run: npm run test:unit
+
+  index-tests:
+    runs-on: ubuntu-latest
+    needs: build
+    env:
+      SFCC_OAUTH_CLIENT_ID: ${{ secrets.SFCC_OAUTH_CLIENT_ID }}
+      SFCC_OAUTH_CLIENT_SECRET: ${{ secrets.SFCC_OAUTH_CLIENT_SECRET }}
+      SANDBOX_HOST: ${{ vars.SANDBOX_HOST }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+
+      - name: Install sfcc-ci
+        run: sudo npm install -g sfcc-ci
+
+      - name: Authenticate with SFCC
+        run: |
+          sfcc-ci client:auth "$SFCC_OAUTH_CLIENT_ID" "$SFCC_OAUTH_CLIENT_SECRET"
+          echo "ACCESS_TOKEN=$(sfcc-ci client:auth:token)" >> $GITHUB_ENV
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Install Algolia Dependencies
+        run: npm install algoliasearch
+
+      - name: Run Algolia Index Tests
+        run: npm test -- test/e2e/variation_index.test.js
```

Comment:
What do you expect this to do? `npm test` runs the legacy tests.
Having GitHub Actions workflow run successfully is one thing, but you need to also look at the logs to ensure that it does what you expect. Here are the logs of the command (from [this run](https://github.com/algolia/algoliasearch-sfcc-b2c/actions/runs/11854239913/job/33036082203)):

```
Run npm test -- test/e2e/variation_index.test.js

> int_algolia_sfra@24.5.0 test
> sgmf-scripts --test test/unit/**/model/algoliaProduct.js test/e2e/variation_index.test.js

/home/runner/work/algoliasearch-sfcc-b2c/algoliasearch-sfcc-b2c/node_modules/.bin/_mocha "test/unit/**/model/algoliaProduct.js" --reporter spec


  algoliaProduct module - Test Algolia Product model
    ✓ Check if the Algolia Product model is valid


  1 passing (8ms)
```
The `variation_index.test.js` is never run.